src/editions/SingleBatchEditionFactory.sol

```diff
@@ -5,16 +5,38 @@ import {ClonesUpgradeable} from "@openzeppelin-contracts-upgradeable/proxy/Clone
 
 import {ISingleBatchEdition} from "nft-editions/interfaces/ISingleBatchEdition.sol";
 import {SingleBatchEdition} from "nft-editions/SingleBatchEdition.sol";
+import {SSTORE2} from "solmate/utils/SSTORE2.sol";
 
 import {IGatedEditionMinter} from "./interfaces/IGatedEditionMinter.sol";
 import {IShowtimeVerifier, Attestation, SignedAttestation} from "src/interfaces/IShowtimeVerifier.sol";
 import "./interfaces/Errors.sol";
 
+
 interface IOwnable {
     function transferOwnership(address newOwner) external;
 }
 
-contract SingleBatchEditionCreator {
+/// @param name Name of the edition contract
+/// @param description Description of the edition entry
+/// @param animationUrl Animation url (optional) of the edition entry
+/// @param imageUrl Metadata: Image url (semi-required) of the edition entry
+/// @param royaltyBPS BPS amount of royalties
+/// @param signedAttestation the attestation to verify along with a corresponding signature
+/// @param externalUrl Metadata: External url (optional) of the edition entry
+/// @param creatorName Metadata: Creator name (optional) of the edition entry
+/// @param tags list of comma-separated tags for this edition, emitted as part of the CreatedBatchEdition event
+struct EditionData {
+    string name;
+    string description;
+    string animationUrl;
+    string imageUrl;
+    uint256 royaltyBPS;
+    string externalUrl;
+    string creatorName;
+    string tags;
+}
```

Comment:
This feels nicer than a long function argument!

Reply:
I don't know why I resisted doing this for so long 😅

Reply:
I think so